### DISCUSSION TITLE
Lazy-load persisted query cache

### DISCUSSION THREAD

**Discussion by dimaip** [TanStack/query#2107] [Q&A]
Hi!
I'm using a solution similar to this to persist query cache to indexeddb: https://github.com/tannerlinsley/react-query/discussions/1638#discussion-1747401

Now the problem is that it stores the whole cache in a single entry and rehydrates it all at once.
What I would rather have is that each query key would be a dedicated entry in persisted cache and that it would be restored lazily on first query call, not all at once.

Is there a way to achieve smth like this with current API or should I go back to managing this persistance logic outside of react-query?
Basically I want in-memory query cache to be like an L1 cache and indexed db to be like an L2 cache. Key not in L1? Check L2. Not in L2? Do the query.

For context: my app is a calendar app where you have a ton of information for each day. It pre-caches things for offline use 30 days ahead. It could add up to quite a bit of data, and trying to restore it in one gulp becomes problematic and unreasonable.

**Comment by dimaip** [comment]
Anyone? 🙏

**Comment by zaguiini** [comment]
That would be great! Bumping it.
